components/renku_data_services/namespace/orm.py

```diff
@@ -71,15 +71,37 @@ class NamespaceORM(BaseORM):
 
     def dump(self) -> models.Namespace:
         """Create a namespace model from the ORM."""
+        if self.group_id is None and self.user_id is None:
+            raise errors.ProgrammingError(
+                message=f"Trying to convert a NamespaceORM with ID {self.id} to a model "
+                "but the namespace is missing both the user and group ID"
+            )
+        if self.group is None and self.user is None:
+            raise errors.ProgrammingError(
+                message=f"Trying to convert a NamespaceORM with ID {self.id} to a model "
+                "but the namespace is missing both the user and group relations"
+            )
         created_by = self.user_id
+        name: str | None = None
         if self.group:
             created_by = self.group.created_by
+            name = self.group.name
+        elif self.user:
+            if self.user.first_name and self.user.last_name:
+                name = f"{self.user.first_name} {self.user.last_name}"
+            else:
+                name = self.user.first_name or self.user.last_name
+        # NOTE: Mypy should have inferred the 2 types below beacuse at the top we check that both values are not None
```

Comment:
To avoid this, you can have:

```python
    def dump(self) -> models.Namespace:
        """Create a namespace model from the ORM."""
        if self.group_id and self.group:
            created_by = self.group.created_by
            name = self.group.name
            underlying_resource_id = self.group_id
            return models.Namespace(
                id=self.id,
                slug=self.slug,
                latest_slug=self.slug,
                created_by=created_by,
                kind=models.NamespaceKind.group,
                underlying_resource_id=underlying_resource_id,
                name=name,
            )

        if not self.user or not self.user_id:
            raise errors.ProgrammingError(message="Found a namespace that has no group or user associated with it.")

        created_by = self.user_id
        name = (
            f"{self.user.first_name} {self.user.last_name}"
            if self.user.first_name and self.user.last_name
            else self.user.first_name or self.user.last_name
        )
        underlying_resource_id = self.user_id
        return models.Namespace(
            id=self.id,
            slug=self.slug,
            latest_slug=self.slug,
            created_by=created_by,
            kind=models.NamespaceKind.user,
            underlying_resource_id=underlying_resource_id,
            name=name,
        )
```

Reply:
Yes this is nicer. I will add it.